File: frontend/src/types/shared-event/shared-event.ts
import type { BaseEntity } from "../common/base-entity";
import type { User } from "../auth/user";

export interface SharedEventParticipant extends BaseEntity {
  sharedEventId: string;
  userId: string;
  user?: User;
  userName?: string;
  joinedAt?: string;
}

export interface SharedEvent extends BaseEntity {
  groupId: string;
  title: string;
  description: string;
  points: number;
  isClosed: boolean;
  closesAt?: string;
  createdByUserId: string;
  createdByUserName?: string;
  participantCount: number;
  hasCurrentUserJoined: boolean;
  participants?: SharedEventParticipant[];
    isPendingRemoval?: boolean;
    removalVoteDeadline?: string;
    quorumRequired?: number;
    removeCount?: number;
    keepCount?: number;
    commentCount?: number;
    imageUrl?: string;
    createdByUserAvatarUrl?: string;
}

export interface CreateSharedEventRequest {
  groupId: string;
  title: string;
  description: string;
  points: number;
  closesAt?: string;
  imageUrl?: string;
  participantUserIds?: string[];
}

export type CreateSharedEventResponse = SharedEvent;

export interface UpdateSharedEventRequest {
  title?: string;
  description?: string;
  points?: number;
  participantUserIds?: string[];
}

export type UpdateSharedEventResponse = SharedEvent;

export type JoinSharedEventResponse = SharedEventParticipant;

export type LeaveSharedEventResponse = void;

export interface UserSharedEvent extends SharedEvent {
  userRole: "organizer" | "participant";
  participantCount: number;
}

File: backend/src/Entities/SharedEvent.cs
using backend.src.Entities.Base;

namespace backend.src.Entities;

public class SharedEvent : Entity
{
    public Guid GroupId { get; set; }
    public Group Group { get; set; } = null!;
    public Guid CreatedByUserId { get; set; }
    public User CreatedByUser { get; set; } = null!;
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public int Points { get; set; }
    public bool IsClosed { get; set; }
    public DateTime? ClosesAt { get; set; }
    public ICollection<SharedEventParticipant> Participants { get; set; } = new List<SharedEventParticipant>();
}
